### PR TITLE
feat(sdk): expose color and typography options for smart scalar in embedding SDK

### LINKED ISSUE #42563: Support extended theming for smart scalar for embedding SDK

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/42563

### Description

Expose the positive and negative colors used in the previous value comparison to the theme settings, as well as the scalar value's font size and line height options.

### How to verify

- Navigate to the `test-sdk-scalar` branch in the demo app
- Note the values defined in `AppProvider.tsx` - you can change the colors and font size for testing.
- Verify that the CSS workarounds in `product-list.css` are deleted
- Go to the product page in the demo. The positive color and value font size/line height should be updated accordingly.

### CODE REVIEW

File: frontend/src/metabase/hooks/use-embedding-theme-options.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { EmbeddingTheme } from "embedding-sdk/types/theme/private";
+import { useMantineTheme } from "metabase/ui";
+
+export const useEmbeddingThemeOptions = () => {
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/hooks/use-embedding-theme-options.ts:4]: @deniskaber Good point! This hook is already in `frontend/src/metabase/hooks` though, do you have a recommended directory to store this hook other than `frontend/src/metabase`? I'm importing the `EmbeddingTheme` type from the SDK (which is the extended `MantineTheme` type) into the main app here.

Another idea I was thinking about is whether we could get rid of `EmbeddingTheme` type and just add the `other` part, and just integrate it to the main app's `MantineTheme`. My worry is mostly that we don't namespace `mantine.other` yet, so other teams might be confused where `theme.other` comes from.

[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/hooks/use-embedding-theme-options.ts:4]: I see! I'll do that then - and we can communicate to other teams about this. I can see `theme.scalar.fontSize` is generic enough to be useful to other teams, thanks 👍🏻

**Review by alxnddr — COMMENTED** [review]
How are we testing these changes? Wdyt about adding stories for both unthemed and themed variants? Then we can have visual tests for them using loki

File: frontend/src/metabase/hooks/use-embedding-theme-options.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/hooks/use-embedding-theme-options.ts]: @deniskaber I got rid of `EmbeddingTheme`, `EmbeddingThemeOverride` and `useEmbeddingThemeOptions` in my latest commit. Now the theme options are available as part of the usual `MantineTheme`, and can be accessed with the usual `useMantineTheme` hook re-exported from Mantine in `metabase/ui`.

I'll re-request your review once I've added the stories as recommended by Sasha. Thanks!

File: loki.config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   diffingEngine: "looks-same",
-  storiesFilter: "static-viz",
+  storiesFilter: "static-viz|viz",
```
Comment (heypoom) [inline loki.config.js:3]: I think the stories filter is a regex, so I'm not sure if we need a `^viz|static-viz` instead

File: frontend/test/__support__/ui.tsx [hunk 3]
```diff
@@ -207,11 +208,11 @@ function SdkWrapper({
   return (
     <Provider store={store}>
       <EmotionCacheProvider>
-        <ThemeProvider>
+        <SdkThemeProvider>
```
Comment (heypoom) [inline frontend/test/__support__/ui.tsx:211]: updated the SDK wrapper to use the `SdkThemeProvider` instead to be more realistic

File: frontend/test/__support__/ui.tsx [hunk 4]
```diff
@@ -157,7 +158,7 @@ export function renderWithProviders(
   };
 }
 
-function Wrapper({
+export function TestWrapper({
```
Comment (heypoom) [inline frontend/test/__support__/ui.tsx:161]: this is also useful for storybook and loki (visual testing) so let's expose it

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 5]
```diff
@@ -285,25 +285,42 @@ const theme = {
...
+
     /** Color used for popover shadows */
     shadow: "rgba(0,0,0,0.08)",
   },
 
-  table: {
-    cell: {
-      // Text color of cells, defaults to `text-dark`
-      textColor: "#4C5773",
+  components: {
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/README.md:298]: noticed that the structure in the current README is wrong, it should be `theme.components.table` not `theme.table`

File: enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx [hunk 6]
```diff
@@ -49,10 +48,11 @@ const SdkContentWrapperInner = styled.div<
   --mb-color-text-medium: ${({ theme }) => theme.fn.themeColor("text-medium")};
   --mb-color-text-light: ${({ theme }) => theme.fn.themeColor("text-light")};
 
+  font-size: ${({ theme }) => theme.other.fontSize ?? "0.875em"};
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx:51]: we can lift `getWrapperStyle` here as `theme.other.fontSize` exists in the global Mantine scope now

**Comment by heypoom** [comment]
Note: the storybook story produces an uncaught runtime error `ResizeObserver loop completed with undelivered notifications`, which is caught by Loki.

**Review by alxnddr — APPROVED** [review]
Looks great, thanks for adding the specs!

Seems like loki job works so the ResizeObserver problem does not block the merge, is that correct? I'll investigate it separately then

**Comment by heypoom** [comment]
@alxnddr Yes, the runtime error is only an issue when regenerating the snapshots, but the Loki job for comparing images should run fine. Thanks a ton!

**Review by deniskaber — APPROVED** [review]

File: frontend/src/types/mantine.d.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import type { EmbeddingThemeOptions } from "embedding-sdk/types/theme/private";
+
+declare module "@mantine/core" {
+  /**
+   * Add more theme options to Mantine's `theme.other` field using this type.
+   *
+   * Refer to [https://v6.mantine.dev/theming/theme-object/#other]
+   **/
+  export interface MantineThemeOther extends EmbeddingThemeOptions {}
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/types/mantine.d.ts:9]: @deniskaber great idea! will add this in

**Pull request merged**